json/json.go

```diff
@@ -471,16 +471,19 @@ func (dec *Decoder) InputOffset() int64 {
 
 // Encoder is documented at https://golang.org/pkg/encoding/json/#Encoder
 type Encoder struct {
-	writer io.Writer
-	prefix string
-	indent string
-	buffer *bytes.Buffer
-	err    error
-	flags  AppendFlags
+	writer          io.Writer
+	prefix          string
+	indent          string
+	buffer          *bytes.Buffer
+	err             error
+	flags           AppendFlags
+	addExtraNewline bool
```

Comment:
Could we represent this in `AppendFlags` instead of adding a boolean field? We can use an unexported constant if it doesn't make sense to expose publicly.

Reply:
I can, however, `AppendFlags` don't seem to expose a `.has` method, only `ParseFlags` do.
I know they're in the end the same thing, so I can probably type cast or do the bitwise & manually, but I assumed this wouldn't be the most clean way to expose this.

Reply:
Would you have any recommendation?

Reply:
Would following the pattern of the other methods work in this case? For example https://github.com/segmentio/encoding/blob/master/json/json.go#L538-L546

Reply:
If the manual bitwise `&` sounds good to you, I believe this commit: [1c1221b](https://github.com/segmentio/encoding/pull/119/commits/1c1221bb565b0bf717bec2f8e3f36917d03f1e47) should do what you suggested.

Reply:
I got confused because `ParseFlags` have an explicit `.has` method that does this bitwise `&` so I expected the lack of it for `AppendFlags` might have meant there was something else needed.